src/OpenTelemetry.Shared/IOfflineStorage.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="IOfflineStorage.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Shared
+{
+    /// <summary>
+    /// Persistent storage API.
+    /// </summary>
+    public interface IOfflineStorage
+    {
+        /// <summary>
+        /// Reads all blobs from the storage.
+        /// </summary>
+        /// <returns>All blobs from storage.</returns>
+        public IEnumerable<IOfflineBlob> GetBlobs();
+
+        /// <summary>
+        /// Reads the latest stored blob from the storage.
+        /// </summary>
+        /// <returns>Blob content.</returns>
+        public IOfflineBlob GetBlob();
+
+        /// <summary>
+        /// Writes a telemetry buffer to storage.
+        /// </summary>
+        /// <param name="buffer">Telemetry buffer.</param>
+        /// <param name="leasePeriod">Lease period in seconds.</param>
+        public void PutBlob(byte[] buffer, int leasePeriod = 0);
```

Comment:
since GetBlob is returning IOfflineBlob, I'd expect PutBlob to accept IOfflineBlog, not byte[]

Reply:
In that case, we need to method signature as follows `public void PutBlob(IOfflineBlob offlineblob, byte[] buffer, int leasePeriod = 0)` But creating a `IOfflineBlob` may not have meaning.

Reply:
public void PutBlob(IOfflineBlob offlineblob, int leasePeriod = 0) -- wont this be enough?

Reply:
I think better option would be to update `GetBlob` to return byte array. Intention of `GetBlob` is to return a blob content. I updated the signature for `GetBlob` which will match the `PutBlob` semantics.

Reply:
I think it should work. Updated it.

Reply:
I don't think this would work.
The semantic of this function needs to be clarified.

Reply:
Thanks Reiley for the call out. Performed research on `Blob` semantics. `PutBlob` functionality should take a parameter which is of content's date type. For example,
* PutBlob(string[] strArr)
* PutBlob(object[] objArr)

So in this case valid parameter would be `byte[] buffer`.
I feel the method signature which I had earlier did not explain the intent correctly. Will update the method summary to `Create a new blob and update content of the created blob.` 

It makes more sense for `GetBlob` to return a blob (`IOfflineBlob`), as the method signature indicates its a `GetBlob` and not `GetBlobContent`.